### PR TITLE
Fast basic test

### PR DESCRIPTION
**Pull request by Sinacam**
This test runs example configs with fake data, ensuring no hard errors occur.
This is relatively fast and is intended to be ran as a github action before merging anything to master.
Also included is several xmlcnn fixes.
This PR depends on #200.

### CODE REVIEW

**Comment by Sinacam** [comment]
`basic_test.yml` is the github actions implementing the test.
The UI below shows how it will work.

File: example_config/rcv1/cnn_tune.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 # data
 training_file: data/rcv1/train.txt
-val_file: data/rcv1/valid.txt
 test_file: data/rcv1/test.txt
 data_name: rcv1
 min_vocab_freq: 1
```
Comment (Eleven1Liu) [inline example_config/rcv1/cnn_tune.yml:5]:
```suggestion
min_vocab_freq: 1
add_special_tokens: false
```

Comment (Eleven1Liu) [reply example_config/rcv1/cnn_tune.yml:5]: Fixed in #201.

**Review by Eleven1Liu — APPROVED** [review]
Tune config is fixed in PR #202 .

**Pull request merged**